applications/ShallowWaterApplication/custom_processes/id_renumbering_process.cpp

```diff
@@ -0,0 +1,177 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "id_renumbering_process.h"
+
+
+namespace Kratos
+{
+
+IdRenumberingProcess::IdRenumberingProcess(Model& rThisModel) : mrModel(rThisModel)
+{
+    mModelPartNames = GetRootModelPartNames();
+}
+
+IdRenumberingProcess::IdRenumberingProcess(Model& rThisModel, const StringVectorType& rModelPartNames)
+ : mrModel(rThisModel)
+{
+    mModelPartNames = rModelPartNames;
+}
+
+void IdRenumberingProcess::RenumberNodes()
+{
+    // The new absolute unique_id
+    IndexType unique_id = 0;
+    mOriginNodesIds.clear();
+
+    // Loop the model parts
+    for (const auto& name : mModelPartNames)
+    {
+        ModelPart& model_part = mrModel.GetModelPart(name);
+        for (int i = 0; i < static_cast<int>(model_part.NumberOfNodes()); ++i)
```

Comment:
If the loops are not OpenMP parallel (and they can't be in this specific case), the counter type can be `unsigned int` so you don't need the `static_cast<int>()` anymore.

Reply:
Fixed